functests/xt_u32/xt_u32.go

```diff
@@ -0,0 +1,223 @@
+package xt_u32
+
+import (
+	"context"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"os"
+	"strings"
+	"time"
+
+	igntypes "github.com/coreos/ignition/config/v2_2/types"
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	"github.com/openshift-kni/cnf-features-deploy/functests/utils/client"
+	"github.com/openshift-kni/cnf-features-deploy/functests/utils/discovery"
+	"github.com/openshift-kni/cnf-features-deploy/functests/utils/execute"
+	"github.com/openshift-kni/cnf-features-deploy/functests/utils/images"
+	"github.com/openshift-kni/cnf-features-deploy/functests/utils/namespaces"
+	utilNodes "github.com/openshift-kni/cnf-features-deploy/functests/utils/nodes"
+	"github.com/openshift-kni/cnf-features-deploy/functests/utils/pods"
+	mcfgv1 "github.com/openshift/machine-config-operator/pkg/apis/machineconfiguration.openshift.io/v1"
+	k8sv1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+const hostnameLabel = "kubernetes.io/hostname"
+const TestNamespace = "test-xt-u32"
+
+const xt_u32LoadPath = "/etc/modules-load.d/xt_u32-load.conf"
+
+var (
+	xt_u32NodeSelector string
+	hasNonCnfWorkers   bool
+)
+
+func init() {
+	roleWorkerCNF := os.Getenv("ROLE_WORKER_CNF")
+	if roleWorkerCNF != "" {
+		xt_u32NodeSelector = fmt.Sprintf("node-role.kubernetes.io/%s=", roleWorkerCNF)
+	}
+
+	hasNonCnfWorkers = true
+	if os.Getenv("XT_U32_TEST_HAS_NON_CNF_WORKERS") == "false" {
+		hasNonCnfWorkers = false
+	}
+}
+
+var _ = Describe("xt_u32", func() {
+	var testNode string
+
+	execute.BeforeAll(func() {
+		err := namespaces.Create(TestNamespace, client.Client)
+		Expect(err).ToNot(HaveOccurred())
+
+		err = namespaces.Clean(TestNamespace, "xt-u32", client.Client)
+		Expect(err).ToNot(HaveOccurred())
+
+		if xt_u32NodeSelector == "" {
+			xt_u32NodeSelector, err = findXT_U32NodeSelector()
+			Expect(err).ToNot(HaveOccurred())
+		}
+	})
+
+	BeforeEach(func() {
+		if !hasNonCnfWorkers {
+			Skip("Skipping as no non-enabled nodes are available")
+		}
+
+		namespaces.Clean(TestNamespace, "xt-u32", client.Client)
+		By("Choosing the test node")
+		nodes, err := client.Client.Nodes().List(context.Background(), metav1.ListOptions{
+			LabelSelector: xt_u32NodeSelector,
+		})
+		Expect(err).ToNot(HaveOccurred())
+
+		filtered, err := utilNodes.MatchingOptionalSelector(nodes.Items)
+		Expect(err).ToNot(HaveOccurred())
+
+		if discovery.Enabled() && len(filtered) == 0 {
+			Skip("Did not find a node without xt_u32 module enabled")
+		} else {
+			Expect(len(filtered)).To(BeNumerically(">", 0))
+		}
+		testNode = filtered[0].ObjectMeta.Labels[hostnameLabel]
+	})
+
+	Context("Validate the module is enabled and works", func() {
+		It("Should create an iptables rule inside a pod that has the module enabled", func() {
+			By("Create a pod")
+			pod := xt_u32TestPod("xt-u32", testNode, TestNamespace, []string{"sleep inf"})
```

Comment:
Since this is a one use, it may make sense to wrap the command in the pod definition function.
Same with the namespace.